react/next/index.tsx

```diff
@@ -0,0 +1,225 @@
+              payload: {
+                slidesToShow: items,
+                deviceType: item,
+                containerWidth,
+                itemWidth,
+                shouldCorrectItemPosition,
+              },
+            })
+          } else {
+            dispatch({
+              type: 'load',
+              payload: {
+                slidesToShow: items,
+                deviceType: item,
+              },
+            })
+          }
+        }
+      })
+    }
+
+    const onResize = (value?: any): void => {
+      setNewState(!(value && !props.infinite))
+    }
+
+    setNewState(false)
+    window.addEventListener('resize', onResize)
+    return () => window.removeEventListener('resize', onResize)
+  }, [])
+
+  const getState = (): any => {
+    return {
+      ...state,
+    }
+  }
+
+  const next = (slidesHavePassed = 0) => {
+    const { nextSlides, nextPosition } = populateNextSlides(
+      state,
+      props,
+      slidesHavePassed
+    )
+    if (nextSlides === undefined || nextPosition === undefined) {
+      return
+    }
+    dispatch({
+      type: 'slide',
+      payload: {
+        transform: nextPosition,
+        currentSlide: nextSlides,
+      },
+    })
+  }
+
+  const previous = (slidesHavePassed = 0) => {
+    const { nextSlides, nextPosition } = populatePreviousSlides(
+      state,
+      props,
+      slidesHavePassed
+    )
+    if (nextSlides === undefined || nextPosition === undefined) {
+      // they can be 0, which goes back to the first slide.
+      return
+    }
+    dispatch({
+      type: 'slide',
+      payload: {
+        transform: nextPosition,
+        currentSlide: nextSlides,
+      },
+    })
+  }
+
+  const goToSlide = (slide: number): void => {
+    const { itemWidth } = state
+    dispatch({
+      type: 'slide',
+      payload: {
+        transform: -(itemWidth * slide),
+        currentSlide: slide,
+      },
+    })
+  }
+
+  const renderLeftArrow = (): React.ReactNode => {
+    const { customLeftArrow } = props
+    return (
+      <Arrow
+        custom={customLeftArrow}
+        direction="left"
+        getState={() => getState()}
+        action={previous}
+      />
+    )
+  }
+
+  const renderRightArrow = (): React.ReactNode => {
+    const { customRightArrow } = props
+    return (
+      <Arrow
+        custom={customRightArrow}
+        getState={() => getState()}
+        direction="right"
+        action={next}
+      />
+    )
+  }
+
+  const renderDotsList = (): React.ReactElement<any> | null => {
+    return (
+      <Dots
+        state={state}
+        props={props}
+        goToSlide={goToSlide}
+        getState={() => getState()}
+      />
+    )
+  }
+
+  const { shouldRenderOnSSR } = getInitialState(state, props)
+  const isLeftEndReach = !(state.currentSlide - props.slidesToSlide! >= 0)
+  const isRightEndReach = !(
+    state.currentSlide + 1 + state.slidesToShow <=
+    state.totalItems
+  )
+
+  const shouldShowArrows =
```

Comment:
In this case, I think separate the clauses in separate variables is better. This method if a bit harsh to undestand